Fix for SandboxClassLoader broke MockK initialization.

### Description

Since 6ab25c0 MockK initialization is broken when using RobolectricTestRunner.

```
java.lang.NoClassDefFoundError: io/mockk/proxy/jvm/dispatcher/JvmMockKWeakMap
	at io.mockk.proxy.jvm.JvmMockKAgentFactory$init$Initializer.handlerMap(JvmMockKAgentFactory.kt:126)
	at io.mockk.proxy.jvm.JvmMockKAgentFactory$init$Initializer.init(JvmMockKAgentFactory.kt:67)
	at io.mockk.proxy.jvm.JvmMockKAgentFactory.init(JvmMockKAgentFactory.kt:130)
	at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:46)
	at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:172)
	at com.szauren.mockk.MockkInitTestCase.setUp(MockkInitTestCase.kt:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:541)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:254)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: io.mockk.proxy.jvm.dispatcher.JvmMockKWeakMap
	at org.robolectric.internal.bytecode.SandboxClassLoader.getByteCode(SandboxClassLoader.java:189)
	at org.robolectric.internal.bytecode.SandboxClassLoader.maybeInstrumentClass(SandboxClassLoader.java:133)
	at org.robolectric.internal.bytecode.SandboxClassLoader.lambda$loadClass$0(SandboxClassLoader.java:119)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.internal.bytecode.SandboxClassLoader.loadClass(SandboxClassLoader.java:119)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 21 more
```

### Steps to Reproduce

1. clone and checkout branch 
 https://github.com/Szauren/robolectric/tree/mockk-integration-tests

2. run gradle task
`./gradlew integration_tests:mockk:test`

### Robolectric & Android Version

`robolectric:4.4-SNAPSHOT`

### Link to a public git repo demonstrating the problem:
https://github.com/Szauren/robolectric/tree/mockk-integration-tests

--- a/build.gradle
+++ b/build.gradle
@@ -12,6 +12,7 @@ buildscript {
         classpath 'com.android.tools.build:gradle:3.5.3'
         classpath 'net.ltgt.gradle:gradle-errorprone-plugin:0.6'
         classpath 'com.netflix.nebula:gradle-aggregate-javadocs-plugin:2.2.1'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.41"
     }
 }
 

--- a/integration_tests/mockk/build.gradle
+++ b/integration_tests/mockk/build.gradle
@@ -1,0 +1,13 @@
+apply plugin: org.robolectric.gradle.RoboJavaModulePlugin
+apply plugin: 'kotlin'
+
+dependencies {
+    api project(":robolectric")
+    compileOnly AndroidSdk.MAX_SDK.coordinates
+
+    testCompileOnly AndroidSdk.MAX_SDK.coordinates
+    testRuntime AndroidSdk.MAX_SDK.coordinates
+    testImplementation "junit:junit:4.12"
+    testImplementation "com.google.truth:truth:0.44"
+    testImplementation 'io.mockk:mockk:1.9.3'
+}

--- a/integration_tests/mockk/src/test/java/org/robolectric/integration_tests/mockk/MockkInitTestCase.kt
+++ b/integration_tests/mockk/src/test/java/org/robolectric/integration_tests/mockk/MockkInitTestCase.kt
@@ -1,0 +1,30 @@
+package org.robolectric.integration_tests.mockk;
+
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+class NumberReturner {
+  fun returnNumber() = 0
+}
+
+@RunWith(RobolectricTestRunner::class)
+class MockkInitTestCase {
+
+  @MockK
+  lateinit var returner: NumberReturner
+
+  @Before
+  fun setUp() = MockKAnnotations.init(this)
+
+  @Test
+  fun `Mockk1`() {
+    every { returner.returnNumber() } returns 1
+    assert(returner.returnNumber() == 1)
+  }
+}
+

--- a/robolectric/src/main/java/org/robolectric/internal/AndroidConfigurer.java
+++ b/robolectric/src/main/java/org/robolectric/internal/AndroidConfigurer.java
@@ -85,6 +85,7 @@ public class AndroidConfigurer {
         .doNotAcquirePackage(
             "scala.") //  run with Maven Surefire (see the RoboSpecs project on github)
         .doNotAcquirePackage("kotlin.")
+        .doNotAcquirePackage("io.mockk.")
         // Fix #958: SQLite native library must be loaded once.
         .doNotAcquirePackage("com.almworks.sqlite4java")
         .doNotAcquirePackage("org.jacoco.");

--- a/sandbox/src/main/java/org/robolectric/JarInstrumentor.java
+++ b/sandbox/src/main/java/org/robolectric/JarInstrumentor.java
@@ -185,6 +185,7 @@ public class JarInstrumentor {
         .doNotAcquirePackage(
             "scala.") //  run with Maven Surefire (see the RoboSpecs project on github)
         .doNotAcquirePackage("kotlin.")
+        .doNotAcquirePackage("io.mockk.")
         // Fix #958: SQLite native library must be loaded once.
         .doNotAcquirePackage("com.almworks.sqlite4java")
         .doNotAcquirePackage("org.jacoco.");

--- a/settings.gradle
+++ b/settings.gradle
@@ -29,3 +29,4 @@ include ':integration_tests:androidx'
 include ':integration_tests:androidx_test'
 include ':integration_tests:ctesque'
 include ':testapp'
+include ":integration_tests:mockk"

